src/test/scala/org/apache/spark/sql/delta/MergeIntoSuiteBase.scala

```diff
@@ -1878,4 +1880,79 @@ abstract class MergeIntoSuiteBase
       (1, 10) // update
     )
   )
+
+  protected def withJobCountListener(jobCount: AtomicInteger)(f: => Unit): Unit = {
+    val listener = new SparkListener {
+      override def onJobStart(jobStart: SparkListenerJobStart): Unit = {
+        // Spark will always log a job start/end event even when the job does not launch any task.
+        if (jobStart.stageInfos.exists(_.numTasks > 0)) {
+          jobCount.incrementAndGet()
+        }
+      }
+    }
+    sparkContext.addSparkListener(listener)
+    sparkContext.listenerBus.waitUntilEmpty(15000)
+    jobCount.set(0)
+
+    Utils.tryWithSafeFinally(f) {
+      sparkContext.listenerBus.waitUntilEmpty(15000)
+      sparkContext.removeSparkListener(listener)
+    }
+  }
+
+  test("findTouchedFiles - no files match target predicate") {
+
+    withTable("source") {
+      append(Seq((1, 10), (2, 20)).toDF("key1", "value1"), "value1")  // target
+      val source = Seq((3, 30)).toDF("key2", "value2")  // source
```

Comment:
why are you using different column names for source and target? if you use the same ones, then you can simply use the insert * and update * . See how the tests with the method `testStar` injects stars using `insert(values = "*")` and `update(set = "*")`.

Reply:
This is simply from the fact that I copied the test from one in `MergeIntoScalaSuite`. I can change it to what you'd like. No big deal.

Reply:
Thank you! appreciate it :) and happy to help if you need more pointers.